components/renku_data_services/crc_migrations/versions/5403953f654f_fix_primary_key_sequences.py

```diff
@@ -0,0 +1,42 @@
+"""fix primary key sequences
+
+Revision ID: 5403953f654f
+Revises: 024b215f9a14
+Create Date: 2023-10-28 13:23:45.947375
+
+"""
+import sqlalchemy as sa
+from alembic import op
+from psycopg import sql
+
+# revision identifiers, used by Alembic.
+revision = "5403953f654f"
+down_revision = "024b215f9a14"
+branch_labels = None
+depends_on = None
+
+
```

Comment:
I'm not sure why this error is happening. Shouldn't a sequence field be unique by default? If no, does it mean that we need to re-run this migration in future again?

Reply:
It is unique and that is the failure. The problem occurs because we did this migration: https://github.com/SwissDataScienceCenter/renku-data-services/blob/main/components/renku_data_services/crc_migrations/versions/0044b54a89b9_migrate_old_crc_tables.py

Here we moved the tables and indices from the public schema to a specific one (called resource_pools) but when we did this we did not move the primary key sequences. And because we did not move these then postgres just automatically created them and it started from 1 rather.